tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2133,6 +2125,13 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
 
   private[emitter] def genLet(name: js.Ident, mutable: Boolean, rhs: js.Tree)(
       implicit outputMode: OutputMode, pos: Position): js.LocalDef = {
+    genLet(name, mutable, Some(rhs))
+  }
+
+  private[emitter] def genLet(name: js.Ident, mutable: Boolean,
+      rhs: Option[js.Tree])(implicit outputMode: OutputMode,
+      pos: Position): js.LocalDef = {
+    require(rhs.isDefined || mutable, "Unset constant is meaningless")
```

Comment:
I have a feeling this invariant can be encoded in the types. Instead of this signature, use instead:
```scala
private[emitter] genEmptyMutableLet(name: js.Ident)(implicit ...)
```
It seems that all call sites statically know whether their `rhs` is empty or not. For the empty case, it further seems that the only call site is https://github.com/gzm0/scala-js/blob/f58bc89165cc9a63dfe382abd0757325c6aea977/tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala#L1013 where `mutable = true`.